### PR TITLE
Hammer CLI Foreman Leapp Plugin v. 0.0.1

### PR DESCRIPTION
**Pull request by stejskalleos**
Requires https://github.com/oamg/foreman_leapp/pull/51

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,32 @@
...
+  https://github.com/oamg/foreman_leapp
+) plugin.
+
+## Versions
+
+This is the list of which version of Foreman Ansible is needed to which version of this plugin.
+
+| hammer_cli_foreman_ansible | 1.0.0 |
+|----------------------------|-------|
+|            foreman_ansible | 1.0.0 |
```
Comment (adamruzicka) [inline README.md]: `s/ansible/leapp` and correct the versions

File: lib/hammer_cli_foreman_leapp.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+require 'hammer_cli'
+require 'hammer_cli_foreman'
+
+require 'hammer_cli_foreman_leapp/version'
+require 'hammer_cli_foreman_leapp/preupgrade_reports'
+
+module HammerCLIForemanLeapp
+  HammerCLI::MainCommand.lazy_subcommand(
+    'preupgrade_reports',
+    'Manage Leapp preupgrade_reports',
```
Comment (adamruzicka) [inline lib/hammer_cli_foreman_leapp.rb]: I guess this is what gets displayed to the user, right? If that is the case, then we should probably drop the underscore from preupgrade_report

File: lib/hammer_cli_foreman_leapp.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+require 'hammer_cli'
+require 'hammer_cli_foreman'
+
+require 'hammer_cli_foreman_leapp/version'
+require 'hammer_cli_foreman_leapp/preupgrade_reports'
+
+module HammerCLIForemanLeapp
+  HammerCLI::MainCommand.lazy_subcommand(
+    'preupgrade_reports',
```
Comment (adamruzicka) [inline lib/hammer_cli_foreman_leapp.rb]: From what I've seen, hammer usually uses kebab-case

File: lib/hammer_cli_foreman_leapp/preupgrade_reports.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+          field :summary, _('Summary'), Fields::LongText
+        end
+      end
+
+      build_options
+    end
+
+    class InfoCommand < HammerCLIForeman::InfoCommand
+      output PreupgradeReportsCommand::ListCommand.output_definition
+      build_options
```
Comment (adamruzicka) [inline lib/hammer_cli_foreman_leapp/preupgrade_reports.rb]: This is a bit misleading. `preupgrade_reports list` shows columns `id` and `job_invocation_id`, however `preupgrade_reports --id $ID` expects `$ID` to be a job invocation id

File: lib/hammer_cli_foreman_leapp/preupgrade_reports.rb [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        collection :preupgrade_report_entries, _('Entries'), hide_blank: true do
+          field :id, _('Id')
+          field :hostname, _('Host name')
+          field :title, _('Title')
+          field :actor, _('Actor')
+          field :audience, _('Audience')
+          field :severity, _('Severity')
+          field :tags, _('Tags'), Fields::List
+          field :summary, _('Summary'), Fields::LongText
+        end
```
Comment (adamruzicka) [inline lib/hammer_cli_foreman_leapp/preupgrade_reports.rb]: This only gets shown in info, right? So maybe it should live in that subcommand and extend list command's output with additional fields?

**Review by adamruzicka — CHANGES_REQUESTED** [review]

File: README.md [hunk 6]
```diff
@@ -1 +1,32 @@
...
+
+This Hammer CLI plugin contains set of commands for [foreman_leapp](
+  https://github.com/oamg/foreman_leapp
+) plugin.
+
+## Versions
+
+This is the list of which version of Foreman Leapp is needed to which version of this plugin.
+
+| hammer_cli_foreman_leapp | 1.0.0 |
```
Comment (ofedoren) [inline README.md:11]: :heart: I'd like to see such approach in all of the hammer plugins. Thanks for being precise! Just don't forget that current hammer-cli-foreman-leapp version is 0.0.1.

File: README.md [hunk 7]
```diff
@@ -1 +1,32 @@
...
+
+    $ mkdir -p ~/.hammer/cli.modules.d/
+
+    $ cat <<EOQ > ~/.hammer/cli.modules.d/foreman_leapp.yml
+    :foreman_leapp:
+      :enable_module: true
+    EOQ
+
+    # to confirm things work, this should return useful output
+    hammer preupgrade_reports --help
```
Comment (ofedoren) [inline README.md]: Should be `hammer preupgrade-reports --help`

**Review by ofedoren — APPROVED** [review]
Looks good so far. Haven't tested functionality though... I've broken my dev env while trying to upgrade to nightlies... But if it works for someone, here's my ACK.

**Comment by ofedoren** [comment]
One addition: after I took a look at the foreman_leapp part, I've noticed that the API doc for the show endpoint has changed. Not sure now if hammer info command will work after cache is updated with that change... Needs to be tested. Hopefully I will manage to re-create dev env to test it in reasonable future.

File: README.md (hunk 6 above)
Comment (stejskalleos) [reply README.md:11]: Thanks! About the version, I'm will update versions when plugins be ready to release.

**Comment by stejskalleos** [comment]
> One addition: after I took a look at the foreman_leapp part, I've noticed that the API doc for the show endpoint has changed. Not sure now if hammer info command will work after cache is updated with that change... Needs to be tested. Hopefully I will manage to re-create dev env to test it in reasonable future.

Yeah for the Leapp API you need to have Leapp branch https://github.com/oamg/foreman_leapp/pull/51

**Comment by stejskalleos** [comment]
Because https://github.com/oamg/foreman_leapp/pull/51 took longer than I was expecting and we need to start with packaging, I removed actions from this plugin so we could release first version and start working on packaging.

**Review by ofedoren — APPROVED** [review]
LGTM :)

**Pull request merged**